packages/babel-types/src/definitions/core.js

```diff
@@ -20,7 +21,7 @@ import defineType, {
 defineType("ArrayExpression", {
   fields: {
     elements: {
-      validate: assertValueType("array")
+      validate: chain(assertValueType("array"), assertEach(assertNodeOrValueType("null", "Expression", "SpreadElement")))
```

Comment:
Just to clear out the confusion for anyone reading this, it is not supposed to be a node of type `NullLiteral`, but an actual `null`, which will result in a hole in the array like this: `var arrayWithHoles = [1, , 3];`. I guess you figured that out because this was merged. I probably should have added an explanatory comment on that line :/